### PR TITLE
OADP-598: Improve datamover backup performance

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
This PR improves the datamover backup performance, the VSB blocking part is transferred from csi plugin to core velero process.
Related CSI plugin PR: https://github.com/openshift/velero-plugin-for-csi/pull/28

### CODE REVIEW

File: pkg/datamover/datamover.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+				tmpVSB := snapmoverv1alpha1.VolumeSnapshotBackup{}
+				err := volumeSnapMoverClient.Get(ctx, kbclient.ObjectKey{Namespace: volumesnapshotbackup.Namespace, Name: volumesnapshotbackup.Name}, &tmpVSB)
+				if err != nil {
+					return false, errors.Wrapf(err, fmt.Sprintf("failed to get volumesnapshotbackup %s/%s", volumesnapshotbackup.Namespace, volumesnapshotbackup.Name))
+				}
+				if len(tmpVSB.Status.Phase) == 0 || tmpVSB.Status.Phase != snapmoverv1alpha1.SnapMoverBackupPhaseCompleted {
+					log.Infof("Waiting for volumesnapshotbackup to complete %s/%s. Retrying in %ds", volumesnapshotbackup.Namespace, volumesnapshotbackup.Name, interval/time.Second)
+					return false, nil
+				}
+
```
Comment (savitharaghunathan) [inline pkg/datamover/datamover.go:70]: Should we explicitly add a log for complete stage?

File: pkg/datamover/datamover.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		velerov1api.BackupNameLabel: backupName,
+	})
+
+	err = volumeSnapMoverClient.List(context.TODO(), &VSBList, VSBListOptions)
+	if err != nil {
+		log.Errorf(err.Error())
+		return err
+	}
+
+	log.Info(fmt.Sprintf("List of all VSBs: %v", VSBList))
```
Comment (dymurray) [inline pkg/datamover/datamover.go]: This is ugly: plz remove

File: pkg/datamover/datamover.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		return nil, err
+	}
+
+	return client, err
+}
+
+func CheckIfVolumeSnapshotBackupsAreComplete(ctx context.Context, volumesnapshotbackups snapmoverv1alpha1.VolumeSnapshotBackupList) error {
+	eg, _ := errgroup.WithContext(ctx)
+	// update this to configurable timeout
+	timeout := 10 * time.Minute
```
Comment (dymurray) [inline pkg/datamover/datamover.go:49]: Open a bug for this please

**Review by dymurray — APPROVED** [review]
Tested and worked for me with backup of multiple volumes and cross-namespace restore. LGTM!

File: pkg/datamover/datamover.go (hunk 3 above)
Comment (shubham-pampattiwar) [reply pkg/datamover/datamover.go:49]: https://issues.redhat.com/browse/OADP-644

**Review by eemcmullan — APPROVED** [review]
Tested this PR and it worked for me as the backup continued while VSB progressed.

**Review by savitharaghunathan — APPROVED** [review]

**Pull request merged**